lib/bolt/util.rb

```diff
@@ -85,6 +85,44 @@ def first_run?
         Bolt::Config.user_path && !File.exist?(first_runs_free)
       end
 
+      def find_file_in_mod(mod, module_file)
+        subdir, relative_path = module_file.split(File::SEPARATOR, 2)
+        case subdir
+        when 'scripts'
+          unless relative_path.nil?
+            mod.file(module_file) || mod.script(File.join(mod, relative_path))
+          else
+            mod.file(module_file)
+          end
+        when 'files'
+          unless relative_path.nil?
+            mod.file(module_file) || mod.file(File.join(mod, relative_path))
+          else
+            mod.file(module_file)
+          end
+        else
+          mod.file(module_file)
+        end
+      end
```

Comment:
This method isn't the easiest to follow. I think this stems from a couple things:

- `module_file` is always split, which results in `subdir` sometimes being the name of a subdir, while other times it ends up being the filename. Just from a naming standpoint, it seems like `subdir` should be `nil` if the path doesn't have at least two parts.
- Nested logic. `||` operators inside `unless` statements inside a `case` statement make this logically a bit hard to follow. Part of the extra complexity comes from checking `relative_path.nil?`, but I'm not sure if there's a way around that. Maybe the way to approach that is with...
- Comments. It would be helpful to have comments describing what filepath is being checked in each statement and why.